src/mcts/search.cc

```diff
@@ -1065,6 +1065,18 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend(
       }
       return NodeToProcess::Collision(node, depth, collision_limit);
     }
+    // probably best place to check for two-fold draws consistently
+    if (node->terminal_type_ == Node::Terminal::TwoFold &&
+                    ( params_.GetTwoFoldDrawLevel() > 0 ) ) {
+      if (params_.GetTwoFoldDrawLevel() == 2 && depth <= 3) {
+        // Level 2: no two-fold draw at depth 3 or lower
+        node->MakeNotTerminal();
+      } else if (params_.GetTwoFoldDrawLevel() == 1) {
+        // Level 1: check whether first repetition was before root
```

Comment:
This is the spot where continuous checks for terminal positions could happen, provided they aren't too costly.